src/main/java/se/bjurr/violations/comments/lib/CommentsCreator.java

```diff
@@ -48,7 +48,22 @@ public static void createComments(
     checkNotNull(commentsProvider, "commentsProvider");
     this.violationsLogger = checkNotNull(violationsLogger, "violationsLogger");
     this.commentsProvider = commentsProvider;
-    files = commentsProvider.getFiles();
+    if (commentsProvider.shouldCommentEveryViolation()) {
```

Comment:
Change all this to just
```
List<Violation> allViolations = violations;
if (!commentsProvider.shouldCommentEveryViolation()) {
 final List<Violation> allViolations = filterChanged(violations);
 files = commentsProvider.getFiles();
}
```

Also change `createComments()` to throw exception if `shouldCreateSingleFileComment && shouldCommentEveryViolation`. Perhaps: `throw new RuntimeException("Cannot comment individual files when including all found violations.")`

Perhaps more things to mention here... I think something like that should work.

Reply:
Maybe I didn't understand something from your code suggestion. But since it currently works so that it only checks changed files. My solution made sure to add all violations into the changed files list, so it would think all of them had changed. Now your suggestion would leave changed files list as empty, which would lead to the program not finding the files the violations belong to and throwing "java.util.NoSuchElementException: No value present". Of course it is possible to be done like this, but that then requires changes to other many other parts to make it work.

Regarding the exception. I only tested with Bitbucket and I don't have experience with Github/Gitlab. But in Bitbucket, you could have individual file comments active, even with all violations. Sure, they wouldn't appear nicely in the changed file in the pull request but instead as individual comments. But I didn't feel like this was a problem, since it is an optional configuration and the user can disable it if they don't like it.

Reply:
I made a proposal: https://github.com/tomasbjerre/violation-comments-lib/pull/12

Reply:
Ok. I tested your proposal and it works fine. I also, during testing understood why you throw an exception. I will now close this pull request, since you have merged yours.

Reply:
There should be new releases of all the plugins now.